src/compositemark.ts

```diff
@@ -137,43 +145,45 @@ add(BOX, (spec: GenericUnitSpec<BOX, Encoding>): LayerSpec => {
     ...continuousFieldType
   };
 
+  const discreteAxisChannelDef = discreteAxisFieldDef !== undefined ? {[discreteAxis]: discreteAxisFieldDef} : {};
```

Comment:
A channelDef is a super class of a fieldDef.
Rename this to `discreteAxisEncodingMixin`?